frontend/src/Event/EventDetails.jsx

```diff
@@ -64,6 +64,32 @@ const EventDetails = ({
     };
   }, [eventId]);
 
+  const handleLaneDataUpdate = (updatedData) => {
```

Comment:
I am not sure about this function.
The function updates `laneData` and `heatData` locally, without syncing or validating these changes with the backend. If another user makes changes, those updates will not automatically reflect.

Reply:
Instead, added a state variable and add it to the useEffect dependency array to trigger the api call to reload the heat and lane details data.